### PR TITLE
write complete task for socket handler

### PR DESCRIPTION
**Pull request by TingDaoK**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by bretambrose — APPROVED** [review]

File: source/socket_channel_handler.c [hunk 1]
```diff
@@ -47,35 +49,120 @@ static int s_socket_process_read_message(
...
-            "static: write of size %llu, completed on channel %p",
-            (unsigned long long)amount_written,
-            (void *)channel);
-
-        if (message->on_completion) {
-            message->on_completion(channel, message, error_code, message->user_data);
-        }
-
         if (socket && socket->handler) {
+            struct socket_write_complete_arg arg = {
```
[the remark this answers is not in the document]
Comment (graebm) [reply source/socket_channel_handler.c:124]: I'm cooking up an alternate fix in this branch: https://github.com/awslabs/aws-c-io/tree/socket-write-complete

Trying to address the issue down one layer, in `posix/socket.c`.

(`windows/iocp/socket.c` was already doing what we wanted, it never fired the callbacks synchronously. Yay Microsoft!)

**Review by graebm — CHANGES_REQUESTED** [review]
Hold off while I try an alternate solution

File: source/socket_channel_handler.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (graebm) [reply source/socket_channel_handler.c:124]: alternate fix: https://github.com/awslabs/aws-c-io/pull/353

**Comment by TingDaoK** [comment]
alternate fix: https://github.com/awslabs/aws-c-io/pull/353

**Pull request closed without merging**